### PR TITLE
Allow creating doc comments around existing comments

### PR DESCRIPTION
**Pull request by davidwengier**
Maybe there is a reason this wasn't supported, and it's a bad idea, but its something that has bit me before because I'm very very silly, and since I saw someone else fall victim to it recently, I thought I'd see how you all feel.

Basically sometimes I write normal comments above my methods, and then on a PR someone tells me to make them XML comments, so I add a `/`, commit and push. And then the build fails because I forgot that XML comments aren't that simple. This fixes that.

Looks like this:
DocCommentsWithComments

### CODE REVIEW

File: src/EditorFeatures/TestUtilities/DocumentationComments/AbstractDocumentationCommentTests.cs [hunk 1]
```diff
@@ -159,13 +159,15 @@ private static Action CreateInsertTextHandler(ITextView textView, string text)
...
-                Assert.Equal(expectedCode, view.TextSnapshot.GetText());
+                var actual = view.TextSnapshot.GetText();
+                Assert.Equal(expectedCode, actual);
 
                 var endCaretPosition = view.Caret.Position.BufferPosition.Position;
-                Assert.True(expectedPosition == endCaretPosition,
-                    string.Format("Caret positioned incorrectly. Should have been {0}, but was {1}.", expectedPosition, endCaretPosition));
+                var actualWithCaret = actual.Insert(endCaretPosition, "$$");
+
+                Assert.Equal(expectedMarkup, actualWithCaret);
```
Comment (davidwengier) [inline src/EditorFeatures/TestUtilities/DocumentationComments/AbstractDocumentationCommentTests.cs:170]: This makes it easier to reason about the caret position failure

File: src/Features/CSharp/Portable/DocumentationComments/CSharpDocumentationCommentSnippetService.cs [hunk 2]
```diff
@@ -256,8 +265,10 @@ protected override bool IsDocCommentNewLine(SyntaxToken token)
         protected override bool IsEndOfLineTrivia(SyntaxTrivia trivia)
             => trivia.RawKind == (int)SyntaxKind.EndOfLineTrivia;
 
-        protected override bool IsSingleExteriorTrivia(DocumentationCommentTriviaSyntax documentationComment, bool allowWhitespace = false)
```
Comment (davidwengier) [inline src/Features/CSharp/Portable/DocumentationComments/CSharpDocumentationCommentSnippetService.cs:259]: Nothing ever passed in a value for `allowWhitespace`

**Comment by CyrusNajmabadi** [comment]
> Maybe there is a reason this wasn't supported, and it's a bad idea

Seems like a great idea to me :)

File: src/Features/CSharp/Portable/DocumentationComments/CSharpDocumentationCommentSnippetService.cs [hunk 3]
```diff
@@ -94,12 +95,20 @@ protected override int GetPrecedingDocumentationCommentCount(MemberDeclarationSy
...
-        protected override List<string> GetDocumentationCommentStubLines(MemberDeclarationSyntax member)
+        protected override List<string> GetDocumentationCommentStubLines(MemberDeclarationSyntax member, string existingCommentText)
         {
+            // Ensure there is at least one space after the ///
+            existingCommentText = existingCommentText switch
+            {
+                { Length: 0 } => " ",
+                [not ' ', ..] => " " + existingCommentText,
+                _ => existingCommentText
+            };
```
Comment (CyrusNajmabadi) [inline src/Features/CSharp/Portable/DocumentationComments/CSharpDocumentationCommentSnippetService.cs:106]: fwiw, i think a conditional of `existingCommentText.StartsWith(' ') ? existingCommentText : " " + existingCommentText` is clearer.  the above took me a lot of time to figure out.

**Review by CyrusNajmabadi — APPROVED** [review]
can we do the same for VB?

**Comment by davidwengier** [comment]
> can we do the same for VB?

I tried, but VB seemed to be surprisingly different as far as parsing doc comments go, so I need to spend more time on that.

Good to at least get the idea validated.

**Pull request merged**